src/app/pages/MarginTradePage/components/ClosedPositionsTable/ClosedPositionRow.tsx

```diff
@@ -1,52 +1,67 @@
 import React, { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { translations } from 'locales/i18n';
 import { TradingPairDictionary } from 'utils/dictionaries/trading-pair-dictionary';
 import { assetByTokenAddress } from 'utils/blockchain/contract-helpers';
 import { TradingPosition } from 'types/trading-position';
-import {
-  toAssetNumberFormat,
-  toNumberFormat,
-  weiToAssetNumberFormat,
-} from 'utils/display-text/format';
-import { Tooltip } from '@blueprintjs/core';
 import { PositionBlock } from '../PositionBlock';
 import { LinkToExplorer } from 'app/components/LinkToExplorer';
-import { AssetRenderer } from 'app/components/AssetRenderer';
 import { TradeProfit } from 'app/components/TradeProfit';
-import { getTradingPositionPrice } from '../../utils/marginUtils';
+import {
+  getClosePositionPrice,
+  getExitTransactionHash,
+  getOpenPositionPrice,
+} from '../../utils/marginUtils';
 import { ActionButton } from 'app/components/Form/ActionButton';
-import { useTranslation } from 'react-i18next';
 import { PositionEventsTable } from '../PositionEventsTable';
-import { translations } from 'locales/i18n';
-import { LoanEvent } from '../OpenPositionsTable/hooks/useMargin_getLoanEvents';
+import { AssetValue } from 'app/components/AssetValue';
+import { toWei } from 'utils/blockchain/math-helpers';
+import { MarginLoansFieldsFragment } from 'utils/graphql/rsk/generated';
 
 type ClosedPositionRowProps = {
-  items: LoanEvent[];
+  event: MarginLoansFieldsFragment;
 };
 
 export const ClosedPositionRow: React.FC<ClosedPositionRowProps> = ({
-  items,
+  event,
 }) => {
   const { t } = useTranslation();
   const [showDetails, setShowDetails] = useState(false);
-  const closedItem = useMemo(() => items[items.length - 1], [items]);
-  const openedItem = useMemo(() => items[0], [items]);
-  const loanAsset = assetByTokenAddress(closedItem.loanToken);
-  const collateralAsset = assetByTokenAddress(closedItem.collateralToken);
+  const {
+    trade,
+    loanToken: { id: loanTokenId },
+    collateralToken: { id: collateralTokenId },
+    liquidates,
+    closeWithSwaps,
+  } = event;
+  const entryLeverage = trade?.[0].entryLeverage || '1';
+  const positionSize = trade?.[0].positionSize || '0';
+  const entryPrice = trade?.[0].entryPrice || '0';
+  const transaction = trade?.[0].transaction.id || '';
```

Comment:
Add these 2 default values to `MarginTradePage/types.ts` as well. 

Or if you think it's better, you can create `DEFAULT_TRADE` object constant there with `entryLeverage`, `positionSize`, `entryPrice` and `transactionId` as properties. I'll leave that up to you.